src/features/events/components/EventItemComponent/index.tsx

```diff
@@ -50,6 +44,20 @@ export const EventItemComponent: FC<Props> = ({
 
   const DescriptionData = () => <Description>{event.description}</Description>
 
+  // Note: the above pattern of creating components for simple syntax
+  // sugar is faulty, as React will identify these as new components always
+  // and therefore create new DOM operations on each render!
+
+  const action = isOwner ? (
+    <Link href={`/events/edit/${event.id}`}>
+      <Button size="small" accent="silent" as="a">
+        Edit
+      </Button>
+    </Link>
+  ) : !isPast ? (
+    <EventActionButton event={event} />
+  ) : null
```

Comment:
This is the remaining of the whole purpose of the `EventActionButton` hub component. It isn't necessarily clear, it could still be improved: but 9 lines of code will certainly hold people shorter to understand.